### PR TITLE
Adding support for non atomic and non CoreOS machines for deploying containerized osds with kv backend

### PR DESCRIPTION
**Pull request by PiotrProkop**

### CODE REVIEW

File: roles/ceph-osd/tasks/docker/start_docker_osd.yml [hunk 1]
```diff
@@ -91,3 +91,19 @@
...
+    image: "{{ ceph_osd_docker_username }}/{{ ceph_osd_docker_imagename }}"
+    name: "{{ ansible_hostname }}-osd-{{ item | regex_replace('/', '') }}"
+    net: host
+    pid: host
+    state: running
+    privileged: yes
+    env: "KV_TYPE={{kv_type}},KV_IP={{kv_endpoint}},OSD_DEVICE={{ item }},{{ ceph_osd_docker_extra_env }}"
+    volumes: "/dev/:/dev/"
+  with_items: ceph_osd_docker_devices
+  when: not is_atomic and ansible_os_family != 'CoreOS'
```
Comment (leseb) [inline roles/ceph-osd/tasks/docker/start_docker_osd.yml]: please use the following syntax:

```
when:
  not is_atomic and
  ansible_os_family != 'CoreOS' and
  osd_containerized_deployment_with_kv
```

**Comment by leseb** [comment]
Would you mind expending some of the `when` conditions? So reading is easier :), thanks!

File: roles/ceph-osd/tasks/docker/start_docker_osd.yml (hunk 1 above)
Comment (PiotrProkop) [inline roles/ceph-osd/tasks/docker/start_docker_osd.yml]: Done.

**Comment by leseb** [comment]
Sorry for the pain but can you stick with this format?

```
 when:
  condition and
 condition2
```
Thanjks!

**Comment by leseb** [comment]
up?

**Comment by PiotrProkop** [comment]
Sorry for delay. I changed them.

**Pull request merged**